src/main/java/de/tum/in/www1/artemis/web/rest/FileUploadExerciseResource.java

```diff
@@ -348,4 +342,32 @@ else if (!authCheckService.isAtLeastTeachingAssistantForExercise(fileUploadExerc
                     "There was an error on the server and the zip file could not be created.")).body(null);
         }
     }
+
+    /**
+     * PUT /file-upload-exercises/re-evaluate : Re-evaluates and updates an existing fileUploadExercise.
+     *
+     * @param fileUploadExercise            the fileUploadExercise to re-evaluate and update
+     * @param deleteFeedbackAfterSGIUpdate  boolean flag that indicates whether the associated feedback should be deleted or not
+     *
+     * @return the ResponseEntity with status 200 (OK) and with body the updated fileUploadExercise, or
+     * with status 400 (Bad Request) if the fileUploadExercise is not valid, or with status 500 (Internal
+     * Server Error) if the fileUploadExercise couldn't be updated
+     * @throws URISyntaxException if the Location URI syntax is incorrect
+     */
+    @PutMapping("/file-upload-exercises/re-evaluate")
+    @PreAuthorize("hasRole('EDITOR')")
+    public ResponseEntity<FileUploadExercise> reEvaluateAndUpdateFileUploadExercise(@RequestBody FileUploadExercise fileUploadExercise,
+            @RequestParam(value = "deleteFeedback", required = false) Boolean deleteFeedbackAfterSGIUpdate) throws URISyntaxException {
+        log.debug("REST request to re-evaluate FileUploadExercise : {}", fileUploadExercise);
+
+        Course course = courseService.retrieveCourseOverExerciseGroupOrCourseId(fileUploadExercise);
+
+        // Check that the user is authorized to update the exercise
+        User user = userRepository.getUserWithGroupsAndAuthorities();
+        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.EDITOR, course, user);
```

Comment:
please fetch the original exercise based on the given id from the database and use it for the authorization check.
Also then make sure that the given exerciseId (as part of the URL) is the same as in the object of the request body